lib/internal/streams/operators.js

```diff
@@ -118,27 +144,26 @@ function map(fn, options) {
             continue;
           }
 
-          if (typeof val?.catch === 'function') {
-            val.catch(onDone);
-          }
+          val = PromiseResolve(val);
+          cnt += 1;
+
 
-          queue.push(val);
+          queue.push(PromisePrototypeThen(val, onThen, onCatch));
```

Comment:
This is wrong. The error won't get thrown when reading.

Reply:
the error will be as:
1. There are tests for that
2. Otherwise you will get an unhandled rejection
3. It will throw as we throw in the `while(true)` loop